File: src/api/reports.ts
import { supabase } from '../lib/supabase';

export interface Report {
  id: string;
  title: string;
  date: string;
  icon: string;
  size: string;
}

export async function getReports(): Promise<Report[]> {
  const { data: { user } } = await supabase.auth.getUser();
  if (!user) throw new Error('Not authenticated');

  const { data, error } = await supabase
    .from('reports')
    .select('*')
    .eq('user_id', user.id)
    .order('report_date', { ascending: false });

  if (error) throw error;
  
  return data.map(row => ({
    id: row.id,
    title: row.title,
    date: row.report_date,
    icon: row.icon,
    size: row.size_label,
  }));
}

export async function shareReport(
  reportId: string,
  target: 'bank' | 'expert',
): Promise<{ success: boolean }> {
  const { data: { user } } = await supabase.auth.getUser();
  if (!user) throw new Error('Not authenticated');

  const { error } = await supabase
    .from('report_shares')
    .insert({ report_id: reportId, user_id: user.id, target });

  if (error) throw error;
  return { success: true };
}

File: backend/app/routers/reports.py
import uuid
from datetime import date

from fastapi import APIRouter, Depends, File, Form, HTTPException, UploadFile, status

from app.config import settings
from app.dependencies import get_current_user_id
from app.db import get_admin_client
from app.schemas.reports import Report, ShareReportRequest, ShareReportResponse
from app.services.seed import ensure_demo_data

router = APIRouter(prefix="/reports", tags=["reports"])


@router.get("", response_model=list[Report])
def get_reports(user_id: str = Depends(get_current_user_id)) -> list[Report]:
    ensure_demo_data(user_id)
    client = get_admin_client()
    rows = (
        client.table("reports")
        .select("*")
        .eq("user_id", user_id)
        .order("report_date", desc=True)
        .execute()
        .data
    )
    return [
        Report(
            id=row["id"],
            title=row["title"],
            date=row["report_date"],
            icon=row["icon"],
            size=row["size_label"],
        )
        for row in rows
    ]


@router.post("/{report_id}/share", response_model=ShareReportResponse)
def share_report(
    report_id: str,
    body: ShareReportRequest,
    user_id: str = Depends(get_current_user_id),
) -> ShareReportResponse:
    ensure_demo_data(user_id)
    client = get_admin_client()
    row = (
        client.table("reports")
        .select("id")
        .eq("id", report_id)
        .eq("user_id", user_id)
        .limit(1)
        .execute()
    )
    if not row.data:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Report not found")

    client.table("report_shares").insert(
        {
            "report_id": report_id,
            "user_id": user_id,
            "target": body.target,
        }
    ).execute()

    return ShareReportResponse(success=True)
